File: services/payment-service/src/models/reconciliation.model.ts
import type { Finding, Severity } from '../domain/reconciliation';

export interface ReconciliationRunRow {
  id: number;
  started_at: Date;
  finished_at: Date | null;
  status: Severity;
  checked_accounts: number;
  drift_cents: number;
  findings: Finding[];
  duration_ms: number | null;
}

export interface ReconciliationRunDto {
  id: number;
  startedAt: Date;
  finishedAt: Date | null;
  status: Severity;
  checkedAccounts: number;
  driftCents: number;
  findings: Finding[];
  durationMs: number | null;
}

export const toReconciliationRunDto = (
  row: ReconciliationRunRow,
): ReconciliationRunDto => ({
  id: row.id,
  startedAt: row.started_at,
  finishedAt: row.finished_at,
  status: row.status,
  checkedAccounts: row.checked_accounts,
  driftCents: row.drift_cents,
  findings: row.findings,
  durationMs: row.duration_ms,
});

File: services/payment-service/src/main/java/com/ledgerflow/payment/models/ReconciliationModel.java
package com.ledgerflow.payment.models;

import com.ledgerflow.payment.domain.Reconciliations.Finding;
import com.ledgerflow.payment.domain.Reconciliations.Severity;
import java.time.Instant;
import java.util.List;

public final class ReconciliationModel {

  private ReconciliationModel() {}

  public record ReconciliationRunRow(
      long id,
      Instant startedAt,
      Instant finishedAt,
      Severity status,
      int checkedAccounts,
      long driftCents,
      List<Finding> findings,
      Integer durationMs) {}

  public record ReconciliationRunDto(
      long id,
      Instant startedAt,
      Instant finishedAt,
      Severity status,
      int checkedAccounts,
      long driftCents,
      List<Finding> findings,
      Integer durationMs) {}

  public static ReconciliationRunDto toReconciliationRunDto(ReconciliationRunRow row) {
    return new ReconciliationRunDto(
        row.id(),
        row.startedAt(),
        row.finishedAt(),
        row.status(),
        row.checkedAccounts(),
        row.driftCents(),
        row.findings(),
        row.durationMs());
  }
}
